lib/config/global.ts

```diff
@@ -65,5 +66,39 @@ export class GlobalConfig {
 
   static reset(): void {
     GlobalConfig.config = {};
+    ExperimentalFlag.reset();
+  }
+}
+export class ExperimentalFlag {
+  private static parsedFlags: Record<string, string> = {};
+
+  static get(key: string): string | null {
+    const experimentalFlags = GlobalConfig.get('experimentalFlags');
+
+    if (!experimentalFlags) {
+      return null;
+    }
+
+    // Check if the flag value is already parsed and stored
+    if (ExperimentalFlag.parsedFlags[key]) {
+      return ExperimentalFlag.parsedFlags[key];
+    }
+
+    for (const flag of experimentalFlags) {
+      if (flag.includes(key)) {
+        const [name, value] = flag.split('=');
+        ExperimentalFlag.parsedFlags[name] = value ?? name;
+        return value ?? name;
+      }
+    }
+
+    return null;
+  }
```

Comment:
thinkig if we should simply add this as `getExperimentalFlag` method to `GlobalConfig` to simpliy it more. 🤔 

WDYT?